env0/utils.go

```diff
@@ -0,0 +1,141 @@
+package env0
+
+import (
+	"errors"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+var matchAllCap = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+// https://stackoverflow.com/questions/56616196/how-to-convert-camel-case-string-to-snake-case
+func toSnakeCase(str string) string {
+	snake := matchFirstCap.ReplaceAllString(str, "${1}_${2}")
+	snake = matchAllCap.ReplaceAllString(snake, "${1}_${2}")
+	return strings.ToLower(snake)
+}
+
+func boolPtr(b bool) *bool {
+	return &b
+}
+
+func stringPtr(s string) *string {
+	return &s
+}
+
+func intPtr(i int) *int {
+	return &i
+}
+
+// Extracts values from the resourcedata, and writes it to the interface.
+func readResourceData(i interface{}, d *schema.ResourceData) error {
+	val := reflect.ValueOf(i).Elem()
+	for i := 0; i < val.NumField(); i++ {
+		fieldName := val.Type().Field(i).Name
+		// Assumes golang is CamalCase and Terraform is snake_case.
+		fieldNameSC := toSnakeCase(fieldName)
+		field := val.Field(i)
+		fieldType := field.Type()
+
+		dval, ok := d.GetOk(fieldNameSC)
+		if !ok {
+			continue
+		}
+
+		switch fieldType.Kind() {
+		case reflect.Ptr:
+			switch field.Type().Elem().Kind() {
+			case reflect.Int:
+				i := dval.(int)
+				field.Set(reflect.ValueOf(&i))
+			case reflect.Bool:
+				b := dval.(bool)
+				field.Set(reflect.ValueOf(&b))
+			}
```

Comment:
Correct me if I'm wrong (probably I am, because your UTs pass) - but aren't we missing here on other primitive data types such as `string`?

Reply:
This is for pointers. Generally, there's no need to do a pointer to string (unlike integers or booleans).
With that said. Yes, you're right. I'm missing many pointer types here. It's on purpose. This is the first iteration, and I would like to improve this mechanism in the future.
I think your suggestion (below) for returning an error on default is a good compromise.

Reply:
https://github.com/env0/terraform-provider-env0/pull/279/commits/a59b4bbbb4c4f0f861546b4848b2c55e5040adbb